sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcColumnarBatchReader.scala

```diff
@@ -0,0 +1,432 @@
+                  index += 1
+                }
+
+              case DecimalType.Fixed(precision, scale) =>
+                val d = fromColumn.asInstanceOf[DecimalColumnVector].vector(0)
+                appendDecimalWritable(toColumn, precision, scale, d)
+
+              case dt =>
+                throw new UnsupportedOperationException(s"Unsupported Data Type: $dt")
+            }
+          }
+        } else if (!field.nullable || fromColumn.noNulls) {
+          field.dataType match {
+            case BooleanType =>
+              val data = fromColumn.asInstanceOf[LongColumnVector].vector
+              var index = 0
+              while (index < batchSize) {
+                toColumn.appendBoolean(data(index) == 1)
+                index += 1
+              }
+
+            case ByteType =>
+              val data = fromColumn.asInstanceOf[LongColumnVector].vector
+              var index = 0
+              while (index < batchSize) {
+                toColumn.appendByte(data(index).toByte)
+                index += 1
+              }
+            case ShortType =>
+              val data = fromColumn.asInstanceOf[LongColumnVector].vector
+              var index = 0
+              while (index < batchSize) {
+                toColumn.appendShort(data(index).toShort)
+                index += 1
+              }
+            case IntegerType | DateType =>
+              val data = fromColumn.asInstanceOf[LongColumnVector].vector
+              var index = 0
+              while (index < batchSize) {
+                toColumn.appendInt(data(index).toInt)
+                index += 1
+              }
+            case LongType =>
+              val data = fromColumn.asInstanceOf[LongColumnVector].vector
+              toColumn.appendLongs(batchSize, data, 0)
+
+            case TimestampType =>
+              val data = fromColumn.asInstanceOf[TimestampColumnVector]
+              var index = 0
+              while (index < batchSize) {
+                toColumn.appendLong(fromTimestampColumnVector(data, index))
+                index += 1
+              }
+
+            case FloatType =>
+              val data = fromColumn.asInstanceOf[DoubleColumnVector].vector
+              var index = 0
+              while (index < batchSize) {
+                toColumn.appendFloat(data(index).toFloat)
+                index += 1
+              }
+            case DoubleType =>
+              val data = fromColumn.asInstanceOf[DoubleColumnVector].vector
+              toColumn.appendDoubles(batchSize, data, 0)
+
+            case StringType =>
+              val data = fromColumn.asInstanceOf[BytesColumnVector]
+              var index = 0
+              while (index < batchSize) {
+                toColumn.appendByteArray(
+                  data.vector(index), data.start(index), data.length(index))
+                index += 1
+              }
+            case BinaryType =>
```

Comment:
Can we merge `StringType` and `BinaryType` into one `case`?

Reply:
Sure.